sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/sql/ToBeamRecord.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql;
+
+import java.util.Map;
+
+import org.apache.beam.sdk.nexmark.model.Event;
+import org.apache.beam.sdk.nexmark.model.KnownSize;
+import org.apache.beam.sdk.nexmark.model.sql.adapter.ModelFieldsAdapter;
+import org.apache.beam.sdk.nexmark.model.sql.adapter.ModelFieldsAdapters;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.BeamRecord;
+
+/**
+ * Convert Java model object to BeamRecord.
+ */
+public class ToBeamRecord {
```

Comment:
<!--thread_id:cc_151138782_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:0-->
<!--section:context-quote-->
> **echauchot** wrote:
> Same as comment about the adapters classes, please read it before. WDYT about including this mechanism it to the Person/Bid/Auction ... classes? Though It might entail the need to create a superclass of the model classes to deduplicate code.

<!--section:body-->
I personally don't think that this would be a good application of inheritance. I think that those data objects are not inherently supposed to be able to convert themselves to BeamRecords, and not all of them might even need to support it, so having an external utility class makes more sense to me.

Reply:
ok, fair enough